system/jlib/jstats.h

```diff
@@ -400,13 +400,18 @@ class jlib_decl StatisticsFilter : public CInterface
 class jlib_decl StatisticsMapping
 {
 public:
-    //Takes a list of StatisticKind
-    StatisticsMapping(const std::initializer_list<StatisticKind> &kinds);
-    //Takes an existing Mapping, and extends it with a list of StatisticKind
-    StatisticsMapping(const StatisticsMapping * from, const std::initializer_list<StatisticKind> &kinds);
-    //Accepts all StatisticKind values
-    StatisticsMapping();
-
+    //Takes a list of StatisticKind and a variable number of existing mappings and combines
+    template <typename... Mappings>
+    StatisticsMapping(const std::initializer_list<StatisticKind> &kinds, const Mappings &... mappings) : StatisticsMapping(&mappings...)
```

Comment:
Changing to a reference makes sense, but why reverse the order, and why inline the function (I guess because it is now a template class?)

Reply:
> why reverse the order

because the tail is a variadic list of existing StatisticMapping's.
e.g. for use case in thor like:
```
const StatisticsMapping joinActivityStatistics({StNumLeftRows, StNumRightRows}, basicActivityStatistics, spillStatistics);
```

>why inline the function (I guess because it is now a template class?)
yes has to be inline because it's a template.

Reply:
Very clever, and painful to my brain.